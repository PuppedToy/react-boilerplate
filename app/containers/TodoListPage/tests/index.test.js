/**
 *
 * Tests for TodoListPage
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { MockedProvider } from '@apollo/react-testing';

import { TodoListPage } from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

import {
  GET_TODOS_QUERY,
  // ADD_TODO_QUERY,
  // DELETE_TODO_QUERY,
  // TOGGLE_TODO_QUERY,
  // EDIT_TODO_QUERY,
} from '../queries';

const mocks = [
  {
    request: {
      query: GET_TODOS_QUERY,
    },
    result: () => [
      {
        id: 1,
        title: 'Item 1',
        done: false,
      },
      {
        id: 2,
        title: 'Item 2',
        done: true,
      },
      {
        id: 3,
        title: 'Item 3',
        done: false,
      },
    ],
  },
];

describe('<TodoListPage />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');

    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <MockedProvider mocks={mocks} addTypename={false}>
          <TodoListPage />
        </MockedProvider>
      </IntlProvider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Expect to load todos', () => {
    const { queryAllByRole } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <MockedProvider mocks={mocks} addTypename={false}>
          <TodoListPage />
        </MockedProvider>
      </IntlProvider>,
    );

    expect(queryAllByRole('button').length).toBe(3);
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <MockedProvider mocks={mocks} addTypename={false}>
          <TodoListPage />
        </MockedProvider>
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
