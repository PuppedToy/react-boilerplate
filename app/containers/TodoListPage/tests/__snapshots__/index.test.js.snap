// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoListPage /> Unit testing with mocked Apollo server Should render and match the snapshot 1`] = `
<div>
  <h1
    class="H1-tncdg6-0 hpAKSC"
  >
    <span>
      To do List
    </span>
  </h1>
  <div>
    <div
      class="Wrapper-sc-12k4hfk-0 bCxcuC"
    >
      <div
        class="Wrapper-ziinx6-0 ecjWnr"
        id="todo-1"
      >
        <div
          class="TodoTitle-xxpseh-0 cHcgZX"
          data-testid="todo-1"
          role="button"
          type="button"
        >
          Item 1
        </div>
        <div
          class="TodoOptions-sc-1s3b1hj-0 jWmTOe"
        >
          <div
            alt="check"
            class="Checkbox-eu7a0t-0 hjQtxn"
            data-testid="toggle-1"
            role="button"
            type="button"
          />
          <div
            alt="delete"
            class="DeleteButton-sc-2zb8s8-0 ceLVod"
            data-testid="delete-1"
            role="button"
            type="button"
          />
        </div>
      </div>
      <div
        class="Wrapper-ziinx6-0 ecjWnr"
        id="todo-2"
      >
        <div
          class="TodoTitle-xxpseh-0 cHcgZX"
          data-testid="todo-2"
          role="button"
          type="button"
        >
          Item 2
        </div>
        <div
          class="TodoOptions-sc-1s3b1hj-0 jWmTOe"
        >
          <div
            alt="uncheck"
            class="Checkbox-eu7a0t-0 hjQtxn"
            data-testid="toggle-2"
            role="button"
            type="button"
          />
          <div
            alt="delete"
            class="DeleteButton-sc-2zb8s8-0 ceLVod"
            data-testid="delete-2"
            role="button"
            type="button"
          />
        </div>
      </div>
      <div
        class="Wrapper-ziinx6-0 ecjWnr"
        id="todo-3"
      >
        <div
          class="TodoTitle-xxpseh-0 cHcgZX"
          data-testid="todo-3"
          role="button"
          type="button"
        >
          Item 3
        </div>
        <div
          class="TodoOptions-sc-1s3b1hj-0 jWmTOe"
        >
          <div
            alt="check"
            class="Checkbox-eu7a0t-0 hjQtxn"
            data-testid="toggle-3"
            role="button"
            type="button"
          />
          <div
            alt="delete"
            class="DeleteButton-sc-2zb8s8-0 ceLVod"
            data-testid="delete-3"
            role="button"
            type="button"
          />
        </div>
      </div>
    </div>
    <div>
      <input
        data-testid="add-todo-input"
        type="text"
        value=""
      />
      <button
        data-testid="add-todo-button"
        type="button"
      >
        <span>
          Add
        </span>
      </button>
    </div>
  </div>
</div>
`;
